ntoskrnl/io/iomgr/driver.c

```diff
@@ -538,6 +538,19 @@ IopInitializeDriverModule(
     if (!(NtHeaders->OptionalHeader.DllCharacteristics & IMAGE_DLLCHARACTERISTICS_WDM_DRIVER))
         driverObject->Flags |= DRVO_LEGACY_DRIVER;
 
+    if (driverObject->Flags & DRVO_LEGACY_DRIVER)
+    {
+       Status = IopCreateLegacyDriverNode(&ServiceName, ServiceHandle);
+       if (!NT_SUCCESS(Status))
+        {
+            ExFreePoolWithTag(nameInfo, TAG_IO); // container for RegistryPath
+            RtlFreeUnicodeString(&ServiceName);
+            RtlFreeUnicodeString(&DriverName);
+            MmUnloadSystemImage(ModuleObject);
```

Comment:
This is interesting, because the other failure paths below leak the module object 😂
The driver object seems to not consistently freed as well.

Reply:
Oh, yeah i'll fix all failure paths.